File: SecondYear/any-mesh_AnimeshKumarSingh_2226CSE(AI&ML)1119_2/Week_4/SecondLargestElement.cpp
#include <bits/stdc++.h> 
int findSecondLargest(int n, vector<int> &arr)
{
    // Write your code here.
    sort(arr.begin(), arr.end());
    int secondHighest = arr[n - 1];
    int highest = arr[n - 1];
    for(int i = n - 2 ; i >= 0 ; i--){
        if(arr[i] < highest){
            secondHighest = arr[i];
            break;
        }
        else if(arr[i] > highest)
            highest = arr[i];
    }
    if(secondHighest == highest)
        return -1;
    return secondHighest;
    
}

File: SecondYear/aayushraghav93_AayushRaghav_2226CSE(AI)1108_2/Week_4/secondlargestelement.java
import java.util.* ;
import java.io.*; 
public class Solution {
	public static int findSecondLargest(int n, int[] arr) {
		// Write your code here.
		 if (arr.length < 2) {
            return -1; 
        }
        int a = Integer.MIN_VALUE;
        int second = Integer.MIN_VALUE;
        
        for (int num : arr) {
            if (num > a) {
                second = a;
                a = num;
            } else if (num > second && num != a) {
                second= num;
            }
        }
        
        if (second == Integer.MIN_VALUE) {
            return -1;
        }
        
        return second;
	}
}
